src/main/java/it/gov/pagopa/pu/workflow/service/debtposition/sync/config/WfExecutionConfigHandlerService.java

```diff
@@ -71,6 +70,25 @@ public void persistAndConfigure(DebtPositionDTO debtPositionDTO, WfExecutionPara
     }
   }
 
+  public FineWfExecutionConfig findStoredExecutionConfig(Long debtPositionId) {
+    WfExecutionConfig config = findWfExecutionConfigById(debtPositionId)
+      .orElseThrow(() -> new InvalidWfExecutionConfigException("Execution config not found for debtPositionId: " + debtPositionId));
+
+    if (!(config instanceof FineWfExecutionConfig fineConfig)) {
+      throw new InvalidWfExecutionConfigException("Invalid execution config type for debtPositionId: " + debtPositionId);
+    }
+
+    return fineConfig;
+  }
```

Comment:
This way this method could be used just for fine WF, instead with the following it could be more general purpose
```suggestion
  public <T extends WfExecutionConfig> T findStoredExecutionConfig(Long debtPositionId, Class<T> wfExecConfigClass) {
    WfExecutionConfig config = findWfExecutionConfigById(debtPositionId)
      .orElseThrow(() -> new InvalidWfExecutionConfigException("Execution config not found for debtPositionId: " + debtPositionId));

    if (config instanceof T requestedConfig)) {
       return requestedConfig;
    } else {
      throw new InvalidWfExecutionConfigException("Invalid execution config type for debtPositionId: " + debtPositionId);
    }
  }
```

Reply:
I can't use instanceOf with generic type T. I pushed an alternative solution using Class<T> and isInstance/cast. Let me know if that works for you